frontend/lib/profile/view_models/profile.viewmodel.dart

```diff
@@ -30,4 +32,51 @@ class ProfileViewModel extends LoadingViewModel {
       notifyListeners();
     }
   }
+
+  Future<bool> updateProfile({
+    required String userId,
+    String? name,
+    String? imageUrl,
+  }) async {
+    if (_profileModel == null) return false;
+    
+    try {
+      _isUpdating = true;
+      notifyListeners();
+
+      // Create optimistic update
+      final updatedProfile = ProfileModel(
+        id: _profileModel!.id,
+        name: name ?? _profileModel!.name,
+        email: _profileModel!.email,
+        imageUrl: imageUrl ?? _profileModel!.imageUrl,
+        rating: _profileModel!.rating,
+        reviews: _profileModel!.reviews,
+        listings: _profileModel!.listings,
+        savedItems: _profileModel!.savedItems,
+      );
+
+      // Update UI immediately
+      _profileModel = updatedProfile;
+      notifyListeners();
+
+      // Make API call
+      final result = await repo.updateProfile(
+        userId: userId,
+        name: name,
+        imageUrl: imageUrl,
+      );
+
+      // Update with server response
+      _profileModel = result;
+      notifyListeners();
```

Comment:
the finally clause always runs before the return, there is no need to call `notifyListeners` here.